mtp_api/apps/transaction/tests/test_bank_admin_views.py

```diff
@@ -82,3 +85,294 @@ def test_create_list(self):
+
+    def _get_authorised_user(self):
+        return self.refund_bank_admins[0]
+
+    def _get_url(self, *args, **kwargs):
+        return reverse('bank_admin:transaction-patch-refunded')
+
+    def _get_transactions(self, tot=30):
+        transactions = generate_transactions(transaction_batch=tot)
+
+        data_list = []
+        for i, trans in enumerate(transactions):
+            trans.save()
+            refund = False
+            if not trans.prisoner_number and not trans.refunded:
+                refund = True
+            data_list.append({'id': trans.id, 'refunded': refund})
+
+        return data_list
+
+    def test_patch_refunded(self):
+        """PATCH on endpoint should update refunded status of given transactions"""
+
+        url = self._get_url()
+        data_list = self._get_transactions()
+
+        user = self._get_authorised_user()
+
+        response = self.client.patch(
+            url, data=data_list, format='json',
+            HTTP_AUTHORIZATION=self.get_http_authorization_for_user(user)
+        )
+        self.assertEqual(response.status_code, http_status.HTTP_200_OK)
+
+        # check changes in db
+        for data in data_list:
+            if data['refunded']:
+                self.assertTrue(Transaction.objects.get(id=data['id']).refunded)
+
+        # check logs
+        refunded_data_list = [t['id'] for t in data_list if t['refunded']]
+        self.assertEqual(
+            Log.objects.filter(
+                user=user,
+                action=LOG_ACTIONS.REFUNDED,
+                transaction__id__in=Transaction.objects.all().values_list('id', flat=True)
+            ).count(),
+            len(refunded_data_list)
+        )
+
+    def _test_patch_invalid_refunds_creates_conflict(self, status):
+        url = self._get_url()
+        user = self._get_authorised_user()
+
+        valid_data_list = self._get_transactions()
+
+        invalid_transactions = Transaction.objects.filter(
+            **Transaction.STATUS_LOOKUP[status])
+        invalid_data_list = (
+            [{'id': t.id, 'refunded': True} for t in invalid_transactions]
+        )
+
+        response = self.client.patch(
+            url, data=valid_data_list + invalid_data_list, format='json',
+            HTTP_AUTHORIZATION=self.get_http_authorization_for_user(user)
+        )
+        self.assertEqual(response.status_code, http_status.HTTP_409_CONFLICT)
+
+        # check that entire update failed
+        for data in valid_data_list:
+            if data['refunded']:
+                self.assertFalse(
+                    Transaction.objects.get(id=data['id']).refunded
+                )
+
+    def test_patch_credited_creates_conflict(self):
+        self._test_patch_invalid_refunds_creates_conflict('credited')
+
+    def test_patch_refunded_creates_conflict(self):
+        self._test_patch_invalid_refunds_creates_conflict('refunded')
+
+    def test_patch_cannot_update_disallowed_fields(self):
```

Comment:
These test names are great; be good to see the body of the tests in here, though, as I said on `_test_create_list_incomplete_data`.